docs/blog/2024-01-31-zephyr-3-5.md

```diff
@@ -0,0 +1,134 @@
+
+### User Config Repositories Using GitHub Actions
+
+Existing user config repositories using Github Actions to build will pull down Zephyr 3.2 automatically, however if you created your user config a while ago, you may need to update it to reference our shared build configuration to leverage the correct Docker image.
+
+1.  Replace the contents of your `.github/workflows/build.yml` with:
+
+    ```yaml
+    on: [push, pull_request, workflow_dispatch]
+
+    jobs:
+      build:
+        uses: zmkfirmware/zmk/.github/workflows/build-user-config.yml@main
+    ```
+
+1.  If it doesn't exist already, add a new file to your repository named `build.yaml`:
+
+    ```yaml
+    # This file generates the GitHub Actions matrix
+    # For simple board + shield combinations, add them
+    # to the top level board and shield arrays, for more
+    # control, add individual board + shield combinations to
+    # the `include` property, e.g:
+    #
+    # board: [ "nice_nano_v2" ]
+    # shield: [ "corne_left", "corne_right" ]
+    # include:
+    #   - board: bdn9_rev2
+    #   - board: nice_nano_v2
+    #     shield: reviung41
+    #
+    ---
+    ```
+
+and then update it as appropriate to build the right shields/boards for your configuration.
+
+### VS Code & Docker (Dev Container)
+
+If you build locally using VS Code & Docker then:
+
+- pull the latest ZMK `main` with `git pull` for your ZMK checkout
+- reload the project
+- if you are prompted to rebuild the remote container, click `Rebuild`
+- otherwise, press `F1` and run `Remote Containers: Rebuild Container`
+- Once the container has rebuilt and reloaded, run `west update` to pull the updated Zephyr version and its dependencies.
+
+Once the container has rebuilt, VS Code will be running the 3.5 Docker image.
+
+### Local Host Development
+
+The following steps will get you building ZMK locally against Zephyr 3.5:
+
+- Run the updated [toolchain installation](/docs/development/setup#toolchain-installation) steps, and once completed, remove the previously installed SDK version (optional, existing SDK should still work)
+- Install the latest version of `west` by running `pip3 install --user --update west`.
+- pull the latest ZMK `main` with `git pull` for your ZMK checkout
+- run `west update` to pull the updated Zephyr version and its dependencies
+
+From there, you should be ready to build as normal!
+
+## Board/Shield Changes
+
+The following changes have [already been completed](https://github.com/zmkfirmware/zmk/pull/1143/commits) for all boards/shields in ZMK `main` branch. For existing or new PRs, or out of tree boards, the following changes are necessary to properly work with the latest changes.
+
+### West Flash Support
+
+If you have a custom board for a target that has a UF2 supporting bootloader, you can easily add support for
+flashing via `west flash`. To do so, add a line to your board's `board.cmake` file like so:
```

Comment:
```suggestion
flashing via `west flash`. (Note that using `west flash` isn't mandatory, you can continue to flash your controllers via the mass storage method.)
To do so, add a line to your board's `board.cmake` file like so:
```
Perhaps worth remarking, just in case someone reading this gets confused?

Reply:
I added a note, but worded it a bit more verbosely to make clear what `west flash` does under the hood. Thoughts?

Reply:
That looks better than mine 👍